routing-daemon/lib/openshift/routing/controllers/batched.rb

```diff
@@ -104,10 +104,10 @@ def create_monitor monitor_name, path, up_code, type, interval, timeout
       @monitors.push monitor_name
     end
 
-    def delete_monitor monitor_name, pool_name=nil
+    def delete_monitor monitor_name, pool_name=nil, type
       raise LBControllerException.new "Monitor not found: #{monitor_name}" unless monitors.include? monitor_name
 
-      @lb_model.delete_monitor monitor_name if monitors.include? monitor_name
+      @lb_model.delete_monitor monitor_name, type if monitors.include? monitor_name
```

Comment:
It's probably a safe assumption given that the type is set in the configuration file and won't be changing at runtime at least.